NationStatesAPIBot/NationStatesAPIBot/Services/DumpDataService.cs

```diff
@@ -1,6 +1,245 @@
-﻿namespace NationStatesAPIBot.Services
+﻿using System;
+using System.Collections.Generic;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Linq;
+using DumpData;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using NationStatesAPIBot.Types;
+
+namespace NationStatesAPIBot.Services
 {
     class DumpDataService
     {
+        private readonly BaseApiService _apiService;
+        private static List<NATION> _nations;
+        private static List<REGION> _regions;
+        private static Task _updating;
+
+        public DumpDataService(ILogger<RecruitmentService> logger, IOptions<AppSettings> appSettings, BaseApiService apiService, CancellationToken cancel)
+        {
+            _apiService = apiService;
+            PeriodicUpdate(TimeSpan.FromDays(1), cancel);
+        }
+        
+        private async Task<List<NATION>> GetNationsAsync()
+        {
+            await Task.WhenAll(_updating);
+            return _nations;
+        }
+        
+        private async Task<List<REGION>> GetRegionsAsync()
+        {
+            await Task.WhenAll(_updating);
+            return _regions;
+        }
+        
+        public async Task PeriodicUpdate(TimeSpan interval, CancellationToken cancel)
+        {
+            while (true)
+            {
+                await Update();
+                await Task.Delay(interval, cancel);
+            }
+        }
+
+        public Task Update()
+        {
+            return _updating = Task.Run(async () => {
```

Comment:
Using await Task.Run isn't necessary.

Reply:
If I don't use Task.Run, how could I best wait for the completion of _updating in GetNationsAsync & GetRegionsAsync?

Reply:
Just by e.g. await GetNationsAsync() it will return when the call is done. I can send you ressources about that if want to. Just let me know.

Reply:
Those will probably come in handy. I don't understand how to wait for a Task without a reference to a Task object.

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/async
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/index